// react-bootstrap-editor/dist/index.esm.js
import {jsx as $QdH60$jsx, jsxs as $QdH60$jsxs, Fragment as $QdH60$Fragment} from "react/jsx-runtime";
import {createRef as $QdH60$createRef, PureComponent as $QdH60$PureComponent} from "react";
import $QdH60$swchelperssrc_define_propertymjs from "@swc/helpers/src/_define_property.mjs";
import {BoldTool as $QdH60$BoldTool, ItalicTool as $QdH60$ItalicTool, UnderlineTool as $QdH60$UnderlineTool, StrikeThroughTool as $QdH60$StrikeThroughTool, H1Tool as $QdH60$H1Tool, H2Tool as $QdH60$H2Tool, H3Tool as $QdH60$H3Tool, FontSizeDownTool as $QdH60$FontSizeDownTool, FontSizeUpTool as $QdH60$FontSizeUpTool, SubscriptTool as $QdH60$SubscriptTool, SuperscriptTool as $QdH60$SuperscriptTool, LinkTool as $QdH60$LinkTool, ForeColorTool as $QdH60$ForeColorTool, BackColorTool as $QdH60$BackColorTool, AlignLeftTool as $QdH60$AlignLeftTool, AlignCenterTool as $QdH60$AlignCenterTool, AlignRightTool as $QdH60$AlignRightTool, AlignFullTool as $QdH60$AlignFullTool, OrderedListTool as $QdH60$OrderedListTool, UnorderedListTool as $QdH60$UnorderedListTool, HorizontalRuleTool as $QdH60$HorizontalRuleTool, IFrameTool as $QdH60$IFrameTool, ImageTool as $QdH60$ImageTool, AudioTool as $QdH60$AudioTool, VideoTool as $QdH60$VideoTool, UndoTool as $QdH60$UndoTool, RedoTool as $QdH60$RedoTool, ResetTool as $QdH60$ResetTool, ClearTool as $QdH60$ClearTool} from "edkit";
import {blobOf as $QdH60$blobOf, parseDOM as $QdH60$parseDOM, walkDOM as $QdH60$walkDOM, insertToCursor as $QdH60$insertToCursor} from "web-utility";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
var $9e8d201707fc0f31$exports = {};

$parcel$export($9e8d201707fc0f31$exports, "renderTool", function () { return $9e8d201707fc0f31$export$56ccecc1dac11363; });


function $9e8d201707fc0f31$export$56ccecc1dac11363(editor) {
    const { title: title , active: active , icon: icon , usable: usable  } = this;
    const Class = `btn btn-${(active ? "" : "outline-") + "secondary"} me-2 mb-2`;
    return /*#__PURE__*/ (0, $QdH60$jsx)("button", {
        type: "button",
        title: title,
        className: Class,
        style: {
            cursor: usable ? "pointer" : "not-allowed"
        },
        disabled: !usable,
        onClick: (event)=>{
            event.preventDefault();
            if (editor.current) this.execute(editor.current);
        },
        children: /*#__PURE__*/ (0, $QdH60$jsx)("i", {
            className: `bi-${icon}`
        })
    }, icon);
}


var $e38b04d9d2c72891$exports = {};

$parcel$export($e38b04d9d2c72891$exports, "TextTools", function () { return $e38b04d9d2c72891$export$e252ba763eff5cdc; });
$parcel$export($e38b04d9d2c72891$exports, "ColorTools", function () { return $e38b04d9d2c72891$export$bdef6e97a10dbad4; });
$parcel$export($e38b04d9d2c72891$exports, "LayoutTools", function () { return $e38b04d9d2c72891$export$145c1b37c0810d7; });
$parcel$export($e38b04d9d2c72891$exports, "MediaTools", function () { return $e38b04d9d2c72891$export$999636c6e25075fb; });
$parcel$export($e38b04d9d2c72891$exports, "ControlTools", function () { return $e38b04d9d2c72891$export$f35db4b82786c3a4; });
$parcel$export($e38b04d9d2c72891$exports, "OriginalTools", function () { return $e38b04d9d2c72891$export$9ddb256264948609; });
$parcel$export($e38b04d9d2c72891$exports, "DefaultTools", function () { return $e38b04d9d2c72891$export$aa718db1466ce312; });
var $84e1677d856194d7$exports = {};

$parcel$export($84e1677d856194d7$exports, "BoldTool", function () { return $84e1677d856194d7$export$10056a49e3b218eb; });
$parcel$export($84e1677d856194d7$exports, "ItalicTool", function () { return $84e1677d856194d7$export$edf99d0f51bacb1e; });
$parcel$export($84e1677d856194d7$exports, "UnderlineTool", function () { return $84e1677d856194d7$export$4b6d9c17188960d2; });
$parcel$export($84e1677d856194d7$exports, "StrikeThroughTool", function () { return $84e1677d856194d7$export$a35bffaeae8fdda0; });
$parcel$export($84e1677d856194d7$exports, "H1Tool", function () { return $84e1677d856194d7$export$30e8c6b428f8749f; });
$parcel$export($84e1677d856194d7$exports, "H2Tool", function () { return $84e1677d856194d7$export$58443375efa91029; });
$parcel$export($84e1677d856194d7$exports, "H3Tool", function () { return $84e1677d856194d7$export$75d24d6b8b747307; });
$parcel$export($84e1677d856194d7$exports, "FontSizeDownTool", function () { return $84e1677d856194d7$export$a20806c6ef6938f8; });
$parcel$export($84e1677d856194d7$exports, "FontSizeUpTool", function () { return $84e1677d856194d7$export$744da7e24e4a7382; });
$parcel$export($84e1677d856194d7$exports, "SubscriptTool", function () { return $84e1677d856194d7$export$6137518b4e998008; });
$parcel$export($84e1677d856194d7$exports, "SuperscriptTool", function () { return $84e1677d856194d7$export$7217f06c61ebd4fc; });
$parcel$export($84e1677d856194d7$exports, "LinkTool", function () { return $84e1677d856194d7$export$f6c54364b5332e66; });



class $84e1677d856194d7$export$10056a49e3b218eb extends (0, $QdH60$BoldTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "type-bold");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $84e1677d856194d7$export$edf99d0f51bacb1e extends (0, $QdH60$ItalicTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "type-italic");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $84e1677d856194d7$export$4b6d9c17188960d2 extends (0, $QdH60$UnderlineTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "type-underline");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $84e1677d856194d7$export$a35bffaeae8fdda0 extends (0, $QdH60$StrikeThroughTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "type-strikethrough");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $84e1677d856194d7$export$30e8c6b428f8749f extends (0, $QdH60$H1Tool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "type-h1");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $84e1677d856194d7$export$58443375efa91029 extends (0, $QdH60$H2Tool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "type-h2");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $84e1677d856194d7$export$75d24d6b8b747307 extends (0, $QdH60$H3Tool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "type-h3");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $84e1677d856194d7$export$a20806c6ef6938f8 extends (0, $QdH60$FontSizeDownTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "sort-alpha-down");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $84e1677d856194d7$export$744da7e24e4a7382 extends (0, $QdH60$FontSizeUpTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "sort-alpha-up");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $84e1677d856194d7$export$6137518b4e998008 extends (0, $QdH60$SubscriptTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "box-arrow-down-right");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $84e1677d856194d7$export$7217f06c61ebd4fc extends (0, $QdH60$SuperscriptTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "box-arrow-up-right");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $84e1677d856194d7$export$f6c54364b5332e66 extends (0, $QdH60$LinkTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "link");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}


var $f79fbcb932e44ba2$exports = {};

$parcel$export($f79fbcb932e44ba2$exports, "ColorSelector", function () { return $f79fbcb932e44ba2$export$9012acaf449cdfcb; });
$parcel$export($f79fbcb932e44ba2$exports, "renderColorTool", function () { return $f79fbcb932e44ba2$export$e913fac2ab7b6f29; });
$parcel$export($f79fbcb932e44ba2$exports, "ForeColorTool", function () { return $f79fbcb932e44ba2$export$cab0e15cfb8489d7; });
$parcel$export($f79fbcb932e44ba2$exports, "BackColorTool", function () { return $f79fbcb932e44ba2$export$92bb6183e7459be2; });




function $f79fbcb932e44ba2$export$9012acaf449cdfcb({ className: className , title: title , type: type , value: value1 , onChange: onChange , icon: icon  }) {
    return /*#__PURE__*/ (0, $QdH60$jsxs)("span", {
        className: `d-inline-block align-middle position-relative ${className}`,
        title: title,
        children: [
            /*#__PURE__*/ (0, $QdH60$jsx)("input", {
                className: "position-absolute w-100 h-100 rounded-3",
                style: {
                    left: 0,
                    top: 0,
                    zIndex: -1
                },
                type: "color",
                value: value1,
                onChange: ({ target: { value: value  }  })=>onChange?.(value)
            }),
            /*#__PURE__*/ (0, $QdH60$jsx)("button", {
                className: "btn",
                style: {
                    color: type === "color" ? value1 : "lightgray",
                    backgroundColor: type === "color" ? "white" : value1,
                    borderColor: value1
                },
                onClick: (event)=>{
                    event.preventDefault();
                    event.currentTarget.previousElementSibling.click();
                },
                children: /*#__PURE__*/ (0, $QdH60$jsx)("i", {
                    className: `bi-${icon}`
                })
            })
        ]
    });
}
function $f79fbcb932e44ba2$export$e913fac2ab7b6f29(editor) {
    const { icon: icon , name: name , colorName: colorName  } = this;
    return /*#__PURE__*/ (0, $QdH60$jsx)($f79fbcb932e44ba2$export$9012acaf449cdfcb, {
        className: "me-2 mb-2",
        title: name,
        icon: icon,
        type: colorName,
        value: this.getColor(),
        onChange: (color)=>editor.current && this.execute(editor.current, color)
    }, icon);
}
class $f79fbcb932e44ba2$export$cab0e15cfb8489d7 extends (0, $QdH60$ForeColorTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "file-earmark-font");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", $f79fbcb932e44ba2$export$e913fac2ab7b6f29);
    }
}
class $f79fbcb932e44ba2$export$92bb6183e7459be2 extends (0, $QdH60$BackColorTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "file-earmark-font-fill");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", $f79fbcb932e44ba2$export$e913fac2ab7b6f29);
    }
}


var $ca6c6c0974c8160e$exports = {};

$parcel$export($ca6c6c0974c8160e$exports, "AlignLeftTool", function () { return $ca6c6c0974c8160e$export$42a444ebfdcfe7b9; });
$parcel$export($ca6c6c0974c8160e$exports, "AlignCenterTool", function () { return $ca6c6c0974c8160e$export$9a2e6e0294a984eb; });
$parcel$export($ca6c6c0974c8160e$exports, "AlignRightTool", function () { return $ca6c6c0974c8160e$export$22363f91860fc258; });
$parcel$export($ca6c6c0974c8160e$exports, "AlignFullTool", function () { return $ca6c6c0974c8160e$export$42827b1362fed748; });
$parcel$export($ca6c6c0974c8160e$exports, "OrderedListTool", function () { return $ca6c6c0974c8160e$export$faf4e70f09f43a15; });
$parcel$export($ca6c6c0974c8160e$exports, "UnorderedListTool", function () { return $ca6c6c0974c8160e$export$cf4030b57a3fbe37; });
$parcel$export($ca6c6c0974c8160e$exports, "HorizontalRuleTool", function () { return $ca6c6c0974c8160e$export$dce8a7d6279fc201; });



class $ca6c6c0974c8160e$export$42a444ebfdcfe7b9 extends (0, $QdH60$AlignLeftTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "text-left");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $ca6c6c0974c8160e$export$9a2e6e0294a984eb extends (0, $QdH60$AlignCenterTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "text-center");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $ca6c6c0974c8160e$export$22363f91860fc258 extends (0, $QdH60$AlignRightTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "text-right");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $ca6c6c0974c8160e$export$42827b1362fed748 extends (0, $QdH60$AlignFullTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "justify");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $ca6c6c0974c8160e$export$faf4e70f09f43a15 extends (0, $QdH60$OrderedListTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "list-ol");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $ca6c6c0974c8160e$export$cf4030b57a3fbe37 extends (0, $QdH60$UnorderedListTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "list-ul");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $ca6c6c0974c8160e$export$dce8a7d6279fc201 extends (0, $QdH60$HorizontalRuleTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "reception-0");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}


var $3ae781e262cfa361$exports = {};

$parcel$export($3ae781e262cfa361$exports, "IFrameTool", function () { return $3ae781e262cfa361$export$3e557574863db086; });
$parcel$export($3ae781e262cfa361$exports, "ImageTool", function () { return $3ae781e262cfa361$export$982f47444a2f88da; });
$parcel$export($3ae781e262cfa361$exports, "AudioTool", function () { return $3ae781e262cfa361$export$6024938f37e646f8; });
$parcel$export($3ae781e262cfa361$exports, "VideoTool", function () { return $3ae781e262cfa361$export$932111b8028f79ed; });



class $3ae781e262cfa361$export$3e557574863db086 extends (0, $QdH60$IFrameTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "window");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $3ae781e262cfa361$export$982f47444a2f88da extends (0, $QdH60$ImageTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "image");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $3ae781e262cfa361$export$6024938f37e646f8 extends (0, $QdH60$AudioTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "voicemail");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $3ae781e262cfa361$export$932111b8028f79ed extends (0, $QdH60$VideoTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "camera-video");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}


var $65ad161bf0bd24b9$exports = {};

$parcel$export($65ad161bf0bd24b9$exports, "UndoTool", function () { return $65ad161bf0bd24b9$export$65235875731ff287; });
$parcel$export($65ad161bf0bd24b9$exports, "RedoTool", function () { return $65ad161bf0bd24b9$export$b60f30d254d1e1a6; });
$parcel$export($65ad161bf0bd24b9$exports, "ResetTool", function () { return $65ad161bf0bd24b9$export$6f46dc3d81fbb4bd; });
$parcel$export($65ad161bf0bd24b9$exports, "ClearTool", function () { return $65ad161bf0bd24b9$export$25eb377ce1feaada; });



class $65ad161bf0bd24b9$export$65235875731ff287 extends (0, $QdH60$UndoTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "arrow-counterclockwise");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $65ad161bf0bd24b9$export$b60f30d254d1e1a6 extends (0, $QdH60$RedoTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "arrow-clockwise");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $65ad161bf0bd24b9$export$6f46dc3d81fbb4bd extends (0, $QdH60$ResetTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "eraser");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}
class $65ad161bf0bd24b9$export$25eb377ce1feaada extends (0, $QdH60$ClearTool) {
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "icon", "file-earmark-x");
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "render", (0, $9e8d201707fc0f31$export$56ccecc1dac11363));
    }
}







const $e38b04d9d2c72891$export$e252ba763eff5cdc = [
    (0, $84e1677d856194d7$export$10056a49e3b218eb),
    (0, $84e1677d856194d7$export$edf99d0f51bacb1e),
    (0, $84e1677d856194d7$export$4b6d9c17188960d2),
    (0, $84e1677d856194d7$export$a35bffaeae8fdda0),
    (0, $84e1677d856194d7$export$30e8c6b428f8749f),
    (0, $84e1677d856194d7$export$58443375efa91029),
    (0, $84e1677d856194d7$export$75d24d6b8b747307),
    (0, $84e1677d856194d7$export$a20806c6ef6938f8),
    (0, $84e1677d856194d7$export$744da7e24e4a7382),
    (0, $84e1677d856194d7$export$6137518b4e998008),
    (0, $84e1677d856194d7$export$7217f06c61ebd4fc),
    (0, $84e1677d856194d7$export$f6c54364b5332e66)
];
const $e38b04d9d2c72891$export$bdef6e97a10dbad4 = [
    (0, $f79fbcb932e44ba2$export$cab0e15cfb8489d7),
    (0, $f79fbcb932e44ba2$export$92bb6183e7459be2)
];
const $e38b04d9d2c72891$export$145c1b37c0810d7 = [
    (0, $ca6c6c0974c8160e$export$42a444ebfdcfe7b9),
    (0, $ca6c6c0974c8160e$export$9a2e6e0294a984eb),
    (0, $ca6c6c0974c8160e$export$22363f91860fc258),
    (0, $ca6c6c0974c8160e$export$42827b1362fed748),
    (0, $ca6c6c0974c8160e$export$faf4e70f09f43a15),
    (0, $ca6c6c0974c8160e$export$cf4030b57a3fbe37),
    (0, $ca6c6c0974c8160e$export$dce8a7d6279fc201)
];
const $e38b04d9d2c72891$export$999636c6e25075fb = [
    (0, $3ae781e262cfa361$export$3e557574863db086),
    (0, $3ae781e262cfa361$export$982f47444a2f88da),
    (0, $3ae781e262cfa361$export$6024938f37e646f8),
    (0, $3ae781e262cfa361$export$932111b8028f79ed)
];
const $e38b04d9d2c72891$export$f35db4b82786c3a4 = [
    (0, $65ad161bf0bd24b9$export$65235875731ff287),
    (0, $65ad161bf0bd24b9$export$b60f30d254d1e1a6),
    (0, $65ad161bf0bd24b9$export$6f46dc3d81fbb4bd),
    (0, $65ad161bf0bd24b9$export$25eb377ce1feaada)
];
const $e38b04d9d2c72891$export$9ddb256264948609 = [
    ...$e38b04d9d2c72891$export$e252ba763eff5cdc,
    ...$e38b04d9d2c72891$export$bdef6e97a10dbad4,
    ...$e38b04d9d2c72891$export$145c1b37c0810d7,
    ...$e38b04d9d2c72891$export$999636c6e25075fb,
    ...$e38b04d9d2c72891$export$f35db4b82786c3a4
];
const $e38b04d9d2c72891$export$aa718db1466ce312 = [
    (0, $84e1677d856194d7$export$10056a49e3b218eb),
    (0, $84e1677d856194d7$export$edf99d0f51bacb1e),
    (0, $84e1677d856194d7$export$4b6d9c17188960d2),
    (0, $84e1677d856194d7$export$a35bffaeae8fdda0),
    (0, $84e1677d856194d7$export$30e8c6b428f8749f),
    (0, $84e1677d856194d7$export$58443375efa91029),
    (0, $84e1677d856194d7$export$75d24d6b8b747307),
    (0, $84e1677d856194d7$export$6137518b4e998008),
    (0, $84e1677d856194d7$export$7217f06c61ebd4fc),
    (0, $f79fbcb932e44ba2$export$cab0e15cfb8489d7),
    (0, $f79fbcb932e44ba2$export$92bb6183e7459be2),
    (0, $ca6c6c0974c8160e$export$42a444ebfdcfe7b9),
    (0, $ca6c6c0974c8160e$export$9a2e6e0294a984eb),
    (0, $ca6c6c0974c8160e$export$22363f91860fc258),
    (0, $ca6c6c0974c8160e$export$42827b1362fed748),
    (0, $ca6c6c0974c8160e$export$faf4e70f09f43a15),
    (0, $ca6c6c0974c8160e$export$cf4030b57a3fbe37),
    (0, $ca6c6c0974c8160e$export$dce8a7d6279fc201),
    (0, $3ae781e262cfa361$export$982f47444a2f88da),
    (0, $65ad161bf0bd24b9$export$65235875731ff287),
    (0, $65ad161bf0bd24b9$export$b60f30d254d1e1a6),
    (0, $65ad161bf0bd24b9$export$25eb377ce1feaada)
];
$parcel$exportWildcard($e38b04d9d2c72891$exports, $84e1677d856194d7$exports);
$parcel$exportWildcard($e38b04d9d2c72891$exports, $f79fbcb932e44ba2$exports);
$parcel$exportWildcard($e38b04d9d2c72891$exports, $ca6c6c0974c8160e$exports);
$parcel$exportWildcard($e38b04d9d2c72891$exports, $3ae781e262cfa361$exports);
$parcel$exportWildcard($e38b04d9d2c72891$exports, $65ad161bf0bd24b9$exports);


var $efab9e4a497112b9$exports = {};

$parcel$export($efab9e4a497112b9$exports, "Editor", function () { return $efab9e4a497112b9$export$7cda8d932e2f33c0; });





class $efab9e4a497112b9$export$7cda8d932e2f33c0 extends (0, $QdH60$PureComponent) {
    static getDerivedStateFromProps({ tools: tools = (0, $e38b04d9d2c72891$export$aa718db1466ce312) , defaultValue: defaultValue  }, { toolList: toolList , data: data  }) {
        return {
            toolList: toolList[0] ? toolList : tools.map((ToolButton)=>new ToolButton()),
            data: data || defaultValue
        };
    }
    componentDidMount() {
        document.addEventListener("selectionchange", this.updateTools);
    }
    componentWillUnmount() {
        document.removeEventListener("selectionchange", this.updateTools);
    }
    async uploadFile(Type, data) {
        if (typeof data === "string" && !/^(data|blob):/.test(data)) return data;
        const tool1 = this.state.toolList.find((tool)=>tool instanceof Type);
        try {
            return await tool1?.save(typeof data === "string" ? await (0, $QdH60$blobOf)(data) : data);
        } catch (error) {
            console.error(error);
            if (typeof data === "string") return data;
        }
    }
    async clearHTML(markup) {
        const fragment = document.createDocumentFragment();
        fragment.append(...(0, $QdH60$parseDOM)(markup));
        for (const element of (0, $QdH60$walkDOM)(fragment, Node.ELEMENT_NODE)){
            const { style: style , dataset: dataset  } = element;
            if ([
                "nowrap",
                "pre"
            ].includes(style.whiteSpace)) style.whiteSpace = "normal";
            for(const key in dataset)delete dataset[key];
            if (element instanceof HTMLImageElement) element.src = await this.uploadFile((0, $3ae781e262cfa361$export$982f47444a2f88da), element.src);
            else if (element instanceof HTMLAudioElement) element.src = await this.uploadFile((0, $3ae781e262cfa361$export$6024938f37e646f8), element.src);
            else if (element instanceof HTMLVideoElement) element.src = await this.uploadFile((0, $3ae781e262cfa361$export$932111b8028f79ed), element.src);
        }
        return fragment;
    }
    render() {
        const { toolList: toolList , data: data  } = this.state, { onChange: onChange  } = this.props;
        return /*#__PURE__*/ (0, $QdH60$jsxs)((0, $QdH60$Fragment), {
            children: [
                /*#__PURE__*/ (0, $QdH60$jsx)("header", {
                    children: toolList.map((tool)=>tool.render(this.box))
                }),
                /*#__PURE__*/ (0, $QdH60$jsx)("div", {
                    ref: this.box,
                    className: "form-control h-auto",
                    contentEditable: true,
                    dangerouslySetInnerHTML: {
                        __html: data
                    },
                    onPaste: this.handlePasteDrop,
                    onDrop: this.handlePasteDrop,
                    onInput: ({ currentTarget: { innerHTML: innerHTML  }  })=>onChange?.(innerHTML)
                })
            ]
        });
    }
    constructor(...args){
        super(...args);
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "box", /*#__PURE__*/ (0, $QdH60$createRef)());
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "state", {
            toolList: [],
            data: ""
        });
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "updateTools", ()=>{
            if (this.box.current === document.activeElement) this.setState({
                toolList: [
                    ...this.state.toolList
                ]
            });
        });
        (0, $QdH60$swchelperssrc_define_propertymjs)(this, "handlePasteDrop", async (event)=>{
            event.preventDefault();
            const { currentTarget: currentTarget  } = event;
            const list = event.type === "paste" ? [
                ...event.clipboardData.items
            ] : [
                ...event.dataTransfer.items
            ];
            for (const item of list)if (item.type === "text/html") {
                const raw = await new Promise((resolve)=>item.getAsString(resolve));
                (0, $QdH60$insertToCursor)(await this.clearHTML(raw));
            } else if (item.type.startsWith("image/")) {
                const URI = await this.uploadFile((0, $3ae781e262cfa361$export$982f47444a2f88da), item.getAsFile());
                if (URI) (0, $QdH60$insertToCursor)(...(0, $QdH60$parseDOM)(`<img src="${URI}" />`));
            } else if (item.type.startsWith("audio/")) {
                const URI = await this.uploadFile((0, $3ae781e262cfa361$export$6024938f37e646f8), item.getAsFile());
                if (URI) (0, $QdH60$insertToCursor)(...(0, $QdH60$parseDOM)(`<audio src="${URI}" />`));
            } else if (item.type.startsWith("video/")) {
                const URI = await this.uploadFile((0, $3ae781e262cfa361$export$932111b8028f79ed), item.getAsFile());
                if (URI) (0, $QdH60$insertToCursor)(...(0, $QdH60$parseDOM)(`<video src="${URI}" />`));
            }
            this.props.onChange?.(currentTarget.innerHTML);
        });
    }
}
(0, $QdH60$swchelperssrc_define_propertymjs)($efab9e4a497112b9$export$7cda8d932e2f33c0, "displayName", "Editor");




export {$9e8d201707fc0f31$export$56ccecc1dac11363 as renderTool, $e38b04d9d2c72891$export$e252ba763eff5cdc as TextTools, $e38b04d9d2c72891$export$bdef6e97a10dbad4 as ColorTools, $e38b04d9d2c72891$export$145c1b37c0810d7 as LayoutTools, $e38b04d9d2c72891$export$999636c6e25075fb as MediaTools, $e38b04d9d2c72891$export$f35db4b82786c3a4 as ControlTools, $e38b04d9d2c72891$export$9ddb256264948609 as OriginalTools, $e38b04d9d2c72891$export$aa718db1466ce312 as DefaultTools, $84e1677d856194d7$export$10056a49e3b218eb as BoldTool, $84e1677d856194d7$export$edf99d0f51bacb1e as ItalicTool, $84e1677d856194d7$export$4b6d9c17188960d2 as UnderlineTool, $84e1677d856194d7$export$a35bffaeae8fdda0 as StrikeThroughTool, $84e1677d856194d7$export$30e8c6b428f8749f as H1Tool, $84e1677d856194d7$export$58443375efa91029 as H2Tool, $84e1677d856194d7$export$75d24d6b8b747307 as H3Tool, $84e1677d856194d7$export$a20806c6ef6938f8 as FontSizeDownTool, $84e1677d856194d7$export$744da7e24e4a7382 as FontSizeUpTool, $84e1677d856194d7$export$6137518b4e998008 as SubscriptTool, $84e1677d856194d7$export$7217f06c61ebd4fc as SuperscriptTool, $84e1677d856194d7$export$f6c54364b5332e66 as LinkTool, $f79fbcb932e44ba2$export$9012acaf449cdfcb as ColorSelector, $f79fbcb932e44ba2$export$e913fac2ab7b6f29 as renderColorTool, $f79fbcb932e44ba2$export$cab0e15cfb8489d7 as ForeColorTool, $f79fbcb932e44ba2$export$92bb6183e7459be2 as BackColorTool, $ca6c6c0974c8160e$export$42a444ebfdcfe7b9 as AlignLeftTool, $ca6c6c0974c8160e$export$9a2e6e0294a984eb as AlignCenterTool, $ca6c6c0974c8160e$export$22363f91860fc258 as AlignRightTool, $ca6c6c0974c8160e$export$42827b1362fed748 as AlignFullTool, $ca6c6c0974c8160e$export$faf4e70f09f43a15 as OrderedListTool, $ca6c6c0974c8160e$export$cf4030b57a3fbe37 as UnorderedListTool, $ca6c6c0974c8160e$export$dce8a7d6279fc201 as HorizontalRuleTool, $3ae781e262cfa361$export$3e557574863db086 as IFrameTool, $3ae781e262cfa361$export$982f47444a2f88da as ImageTool, $3ae781e262cfa361$export$6024938f37e646f8 as AudioTool, $3ae781e262cfa361$export$932111b8028f79ed as VideoTool, $65ad161bf0bd24b9$export$65235875731ff287 as UndoTool, $65ad161bf0bd24b9$export$b60f30d254d1e1a6 as RedoTool, $65ad161bf0bd24b9$export$6f46dc3d81fbb4bd as ResetTool, $65ad161bf0bd24b9$export$25eb377ce1feaada as ClearTool, $efab9e4a497112b9$export$7cda8d932e2f33c0 as Editor};
