// react-bootstrap-editor/dist/index.js
var $4Pi8L$reactjsxruntime = require("react/jsx-runtime");
var $4Pi8L$react = require("react");
var $4Pi8L$swchelperslib_define_propertyjs = require("@swc/helpers/lib/_define_property.js");
var $4Pi8L$edkit = require("edkit");
var $4Pi8L$webutility = require("web-utility");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
var $783b6726ddd1ea56$exports = {};

$parcel$export($783b6726ddd1ea56$exports, "renderTool", function () { return $783b6726ddd1ea56$export$56ccecc1dac11363; });


function $783b6726ddd1ea56$export$56ccecc1dac11363(editor) {
    const { title: title , active: active , icon: icon , usable: usable  } = this;
    const Class = `btn btn-${(active ? "" : "outline-") + "secondary"} me-2 mb-2`;
    return /*#__PURE__*/ (0, $4Pi8L$reactjsxruntime.jsx)("button", {
        type: "button",
        title: title,
        className: Class,
        style: {
            cursor: usable ? "pointer" : "not-allowed"
        },
        disabled: !usable,
        onClick: (event)=>{
            event.preventDefault();
            if (editor.current) this.execute(editor.current);
        },
        children: /*#__PURE__*/ (0, $4Pi8L$reactjsxruntime.jsx)("i", {
            className: `bi-${icon}`
        })
    }, icon);
}


var $ee8be56859ec8fb7$exports = {};

$parcel$export($ee8be56859ec8fb7$exports, "TextTools", function () { return $ee8be56859ec8fb7$export$e252ba763eff5cdc; });
$parcel$export($ee8be56859ec8fb7$exports, "ColorTools", function () { return $ee8be56859ec8fb7$export$bdef6e97a10dbad4; });
$parcel$export($ee8be56859ec8fb7$exports, "LayoutTools", function () { return $ee8be56859ec8fb7$export$145c1b37c0810d7; });
$parcel$export($ee8be56859ec8fb7$exports, "MediaTools", function () { return $ee8be56859ec8fb7$export$999636c6e25075fb; });
$parcel$export($ee8be56859ec8fb7$exports, "ControlTools", function () { return $ee8be56859ec8fb7$export$f35db4b82786c3a4; });
$parcel$export($ee8be56859ec8fb7$exports, "OriginalTools", function () { return $ee8be56859ec8fb7$export$9ddb256264948609; });
$parcel$export($ee8be56859ec8fb7$exports, "DefaultTools", function () { return $ee8be56859ec8fb7$export$aa718db1466ce312; });
var $9a2bbe5526331901$exports = {};

$parcel$export($9a2bbe5526331901$exports, "BoldTool", function () { return $9a2bbe5526331901$export$10056a49e3b218eb; });
$parcel$export($9a2bbe5526331901$exports, "ItalicTool", function () { return $9a2bbe5526331901$export$edf99d0f51bacb1e; });
$parcel$export($9a2bbe5526331901$exports, "UnderlineTool", function () { return $9a2bbe5526331901$export$4b6d9c17188960d2; });
$parcel$export($9a2bbe5526331901$exports, "StrikeThroughTool", function () { return $9a2bbe5526331901$export$a35bffaeae8fdda0; });
$parcel$export($9a2bbe5526331901$exports, "H1Tool", function () { return $9a2bbe5526331901$export$30e8c6b428f8749f; });
$parcel$export($9a2bbe5526331901$exports, "H2Tool", function () { return $9a2bbe5526331901$export$58443375efa91029; });
$parcel$export($9a2bbe5526331901$exports, "H3Tool", function () { return $9a2bbe5526331901$export$75d24d6b8b747307; });
$parcel$export($9a2bbe5526331901$exports, "FontSizeDownTool", function () { return $9a2bbe5526331901$export$a20806c6ef6938f8; });
$parcel$export($9a2bbe5526331901$exports, "FontSizeUpTool", function () { return $9a2bbe5526331901$export$744da7e24e4a7382; });
$parcel$export($9a2bbe5526331901$exports, "SubscriptTool", function () { return $9a2bbe5526331901$export$6137518b4e998008; });
$parcel$export($9a2bbe5526331901$exports, "SuperscriptTool", function () { return $9a2bbe5526331901$export$7217f06c61ebd4fc; });
$parcel$export($9a2bbe5526331901$exports, "LinkTool", function () { return $9a2bbe5526331901$export$f6c54364b5332e66; });



class $9a2bbe5526331901$export$10056a49e3b218eb extends (0, $4Pi8L$edkit.BoldTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "type-bold");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $9a2bbe5526331901$export$edf99d0f51bacb1e extends (0, $4Pi8L$edkit.ItalicTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "type-italic");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $9a2bbe5526331901$export$4b6d9c17188960d2 extends (0, $4Pi8L$edkit.UnderlineTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "type-underline");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $9a2bbe5526331901$export$a35bffaeae8fdda0 extends (0, $4Pi8L$edkit.StrikeThroughTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "type-strikethrough");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $9a2bbe5526331901$export$30e8c6b428f8749f extends (0, $4Pi8L$edkit.H1Tool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "type-h1");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $9a2bbe5526331901$export$58443375efa91029 extends (0, $4Pi8L$edkit.H2Tool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "type-h2");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $9a2bbe5526331901$export$75d24d6b8b747307 extends (0, $4Pi8L$edkit.H3Tool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "type-h3");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $9a2bbe5526331901$export$a20806c6ef6938f8 extends (0, $4Pi8L$edkit.FontSizeDownTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "sort-alpha-down");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $9a2bbe5526331901$export$744da7e24e4a7382 extends (0, $4Pi8L$edkit.FontSizeUpTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "sort-alpha-up");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $9a2bbe5526331901$export$6137518b4e998008 extends (0, $4Pi8L$edkit.SubscriptTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "box-arrow-down-right");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $9a2bbe5526331901$export$7217f06c61ebd4fc extends (0, $4Pi8L$edkit.SuperscriptTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "box-arrow-up-right");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $9a2bbe5526331901$export$f6c54364b5332e66 extends (0, $4Pi8L$edkit.LinkTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "link");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}


var $4c1cbb15589101a6$exports = {};

$parcel$export($4c1cbb15589101a6$exports, "ColorSelector", function () { return $4c1cbb15589101a6$export$9012acaf449cdfcb; });
$parcel$export($4c1cbb15589101a6$exports, "renderColorTool", function () { return $4c1cbb15589101a6$export$e913fac2ab7b6f29; });
$parcel$export($4c1cbb15589101a6$exports, "ForeColorTool", function () { return $4c1cbb15589101a6$export$cab0e15cfb8489d7; });
$parcel$export($4c1cbb15589101a6$exports, "BackColorTool", function () { return $4c1cbb15589101a6$export$92bb6183e7459be2; });




function $4c1cbb15589101a6$export$9012acaf449cdfcb({ className: className , title: title , type: type , value: value1 , onChange: onChange , icon: icon  }) {
    return /*#__PURE__*/ (0, $4Pi8L$reactjsxruntime.jsxs)("span", {
        className: `d-inline-block align-middle position-relative ${className}`,
        title: title,
        children: [
            /*#__PURE__*/ (0, $4Pi8L$reactjsxruntime.jsx)("input", {
                className: "position-absolute w-100 h-100 rounded-3",
                style: {
                    left: 0,
                    top: 0,
                    zIndex: -1
                },
                type: "color",
                value: value1,
                onChange: ({ target: { value: value  }  })=>{
                    return onChange === null || onChange === void 0 ? void 0 : onChange(value);
                }
            }),
            /*#__PURE__*/ (0, $4Pi8L$reactjsxruntime.jsx)("button", {
                className: "btn",
                style: {
                    color: type === "color" ? value1 : "lightgray",
                    backgroundColor: type === "color" ? "white" : value1,
                    borderColor: value1
                },
                onClick: (event)=>{
                    event.preventDefault();
                    event.currentTarget.previousElementSibling.click();
                },
                children: /*#__PURE__*/ (0, $4Pi8L$reactjsxruntime.jsx)("i", {
                    className: `bi-${icon}`
                })
            })
        ]
    });
}
function $4c1cbb15589101a6$export$e913fac2ab7b6f29(editor) {
    const { icon: icon , name: name , colorName: colorName  } = this;
    return /*#__PURE__*/ (0, $4Pi8L$reactjsxruntime.jsx)($4c1cbb15589101a6$export$9012acaf449cdfcb, {
        className: "me-2 mb-2",
        title: name,
        icon: icon,
        type: colorName,
        value: this.getColor(),
        onChange: (color)=>editor.current && this.execute(editor.current, color)
    }, icon);
}
class $4c1cbb15589101a6$export$cab0e15cfb8489d7 extends (0, $4Pi8L$edkit.ForeColorTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "file-earmark-font");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", $4c1cbb15589101a6$export$e913fac2ab7b6f29);
    }
}
class $4c1cbb15589101a6$export$92bb6183e7459be2 extends (0, $4Pi8L$edkit.BackColorTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "file-earmark-font-fill");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", $4c1cbb15589101a6$export$e913fac2ab7b6f29);
    }
}


var $2bb487cfecefdbac$exports = {};

$parcel$export($2bb487cfecefdbac$exports, "AlignLeftTool", function () { return $2bb487cfecefdbac$export$42a444ebfdcfe7b9; });
$parcel$export($2bb487cfecefdbac$exports, "AlignCenterTool", function () { return $2bb487cfecefdbac$export$9a2e6e0294a984eb; });
$parcel$export($2bb487cfecefdbac$exports, "AlignRightTool", function () { return $2bb487cfecefdbac$export$22363f91860fc258; });
$parcel$export($2bb487cfecefdbac$exports, "AlignFullTool", function () { return $2bb487cfecefdbac$export$42827b1362fed748; });
$parcel$export($2bb487cfecefdbac$exports, "OrderedListTool", function () { return $2bb487cfecefdbac$export$faf4e70f09f43a15; });
$parcel$export($2bb487cfecefdbac$exports, "UnorderedListTool", function () { return $2bb487cfecefdbac$export$cf4030b57a3fbe37; });
$parcel$export($2bb487cfecefdbac$exports, "HorizontalRuleTool", function () { return $2bb487cfecefdbac$export$dce8a7d6279fc201; });



class $2bb487cfecefdbac$export$42a444ebfdcfe7b9 extends (0, $4Pi8L$edkit.AlignLeftTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "text-left");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $2bb487cfecefdbac$export$9a2e6e0294a984eb extends (0, $4Pi8L$edkit.AlignCenterTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "text-center");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $2bb487cfecefdbac$export$22363f91860fc258 extends (0, $4Pi8L$edkit.AlignRightTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "text-right");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $2bb487cfecefdbac$export$42827b1362fed748 extends (0, $4Pi8L$edkit.AlignFullTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "justify");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $2bb487cfecefdbac$export$faf4e70f09f43a15 extends (0, $4Pi8L$edkit.OrderedListTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "list-ol");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $2bb487cfecefdbac$export$cf4030b57a3fbe37 extends (0, $4Pi8L$edkit.UnorderedListTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "list-ul");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $2bb487cfecefdbac$export$dce8a7d6279fc201 extends (0, $4Pi8L$edkit.HorizontalRuleTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "reception-0");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}


var $ee1dbf60aeaab96b$exports = {};

$parcel$export($ee1dbf60aeaab96b$exports, "IFrameTool", function () { return $ee1dbf60aeaab96b$export$3e557574863db086; });
$parcel$export($ee1dbf60aeaab96b$exports, "ImageTool", function () { return $ee1dbf60aeaab96b$export$982f47444a2f88da; });
$parcel$export($ee1dbf60aeaab96b$exports, "AudioTool", function () { return $ee1dbf60aeaab96b$export$6024938f37e646f8; });
$parcel$export($ee1dbf60aeaab96b$exports, "VideoTool", function () { return $ee1dbf60aeaab96b$export$932111b8028f79ed; });



class $ee1dbf60aeaab96b$export$3e557574863db086 extends (0, $4Pi8L$edkit.IFrameTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "window");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $ee1dbf60aeaab96b$export$982f47444a2f88da extends (0, $4Pi8L$edkit.ImageTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "image");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $ee1dbf60aeaab96b$export$6024938f37e646f8 extends (0, $4Pi8L$edkit.AudioTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "voicemail");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $ee1dbf60aeaab96b$export$932111b8028f79ed extends (0, $4Pi8L$edkit.VideoTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "camera-video");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}


var $39dd722b123bef71$exports = {};

$parcel$export($39dd722b123bef71$exports, "UndoTool", function () { return $39dd722b123bef71$export$65235875731ff287; });
$parcel$export($39dd722b123bef71$exports, "RedoTool", function () { return $39dd722b123bef71$export$b60f30d254d1e1a6; });
$parcel$export($39dd722b123bef71$exports, "ResetTool", function () { return $39dd722b123bef71$export$6f46dc3d81fbb4bd; });
$parcel$export($39dd722b123bef71$exports, "ClearTool", function () { return $39dd722b123bef71$export$25eb377ce1feaada; });



class $39dd722b123bef71$export$65235875731ff287 extends (0, $4Pi8L$edkit.UndoTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "arrow-counterclockwise");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $39dd722b123bef71$export$b60f30d254d1e1a6 extends (0, $4Pi8L$edkit.RedoTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "arrow-clockwise");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $39dd722b123bef71$export$6f46dc3d81fbb4bd extends (0, $4Pi8L$edkit.ResetTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "eraser");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}
class $39dd722b123bef71$export$25eb377ce1feaada extends (0, $4Pi8L$edkit.ClearTool) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "icon", "file-earmark-x");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "render", (0, $783b6726ddd1ea56$export$56ccecc1dac11363));
    }
}







const $ee8be56859ec8fb7$export$e252ba763eff5cdc = [
    (0, $9a2bbe5526331901$export$10056a49e3b218eb),
    (0, $9a2bbe5526331901$export$edf99d0f51bacb1e),
    (0, $9a2bbe5526331901$export$4b6d9c17188960d2),
    (0, $9a2bbe5526331901$export$a35bffaeae8fdda0),
    (0, $9a2bbe5526331901$export$30e8c6b428f8749f),
    (0, $9a2bbe5526331901$export$58443375efa91029),
    (0, $9a2bbe5526331901$export$75d24d6b8b747307),
    (0, $9a2bbe5526331901$export$a20806c6ef6938f8),
    (0, $9a2bbe5526331901$export$744da7e24e4a7382),
    (0, $9a2bbe5526331901$export$6137518b4e998008),
    (0, $9a2bbe5526331901$export$7217f06c61ebd4fc),
    (0, $9a2bbe5526331901$export$f6c54364b5332e66)
];
const $ee8be56859ec8fb7$export$bdef6e97a10dbad4 = [
    (0, $4c1cbb15589101a6$export$cab0e15cfb8489d7),
    (0, $4c1cbb15589101a6$export$92bb6183e7459be2)
];
const $ee8be56859ec8fb7$export$145c1b37c0810d7 = [
    (0, $2bb487cfecefdbac$export$42a444ebfdcfe7b9),
    (0, $2bb487cfecefdbac$export$9a2e6e0294a984eb),
    (0, $2bb487cfecefdbac$export$22363f91860fc258),
    (0, $2bb487cfecefdbac$export$42827b1362fed748),
    (0, $2bb487cfecefdbac$export$faf4e70f09f43a15),
    (0, $2bb487cfecefdbac$export$cf4030b57a3fbe37),
    (0, $2bb487cfecefdbac$export$dce8a7d6279fc201)
];
const $ee8be56859ec8fb7$export$999636c6e25075fb = [
    (0, $ee1dbf60aeaab96b$export$3e557574863db086),
    (0, $ee1dbf60aeaab96b$export$982f47444a2f88da),
    (0, $ee1dbf60aeaab96b$export$6024938f37e646f8),
    (0, $ee1dbf60aeaab96b$export$932111b8028f79ed)
];
const $ee8be56859ec8fb7$export$f35db4b82786c3a4 = [
    (0, $39dd722b123bef71$export$65235875731ff287),
    (0, $39dd722b123bef71$export$b60f30d254d1e1a6),
    (0, $39dd722b123bef71$export$6f46dc3d81fbb4bd),
    (0, $39dd722b123bef71$export$25eb377ce1feaada)
];
const $ee8be56859ec8fb7$export$9ddb256264948609 = [
    ...$ee8be56859ec8fb7$export$e252ba763eff5cdc,
    ...$ee8be56859ec8fb7$export$bdef6e97a10dbad4,
    ...$ee8be56859ec8fb7$export$145c1b37c0810d7,
    ...$ee8be56859ec8fb7$export$999636c6e25075fb,
    ...$ee8be56859ec8fb7$export$f35db4b82786c3a4
];
const $ee8be56859ec8fb7$export$aa718db1466ce312 = [
    (0, $9a2bbe5526331901$export$10056a49e3b218eb),
    (0, $9a2bbe5526331901$export$edf99d0f51bacb1e),
    (0, $9a2bbe5526331901$export$4b6d9c17188960d2),
    (0, $9a2bbe5526331901$export$a35bffaeae8fdda0),
    (0, $9a2bbe5526331901$export$30e8c6b428f8749f),
    (0, $9a2bbe5526331901$export$58443375efa91029),
    (0, $9a2bbe5526331901$export$75d24d6b8b747307),
    (0, $9a2bbe5526331901$export$6137518b4e998008),
    (0, $9a2bbe5526331901$export$7217f06c61ebd4fc),
    (0, $4c1cbb15589101a6$export$cab0e15cfb8489d7),
    (0, $4c1cbb15589101a6$export$92bb6183e7459be2),
    (0, $2bb487cfecefdbac$export$42a444ebfdcfe7b9),
    (0, $2bb487cfecefdbac$export$9a2e6e0294a984eb),
    (0, $2bb487cfecefdbac$export$22363f91860fc258),
    (0, $2bb487cfecefdbac$export$42827b1362fed748),
    (0, $2bb487cfecefdbac$export$faf4e70f09f43a15),
    (0, $2bb487cfecefdbac$export$cf4030b57a3fbe37),
    (0, $2bb487cfecefdbac$export$dce8a7d6279fc201),
    (0, $ee1dbf60aeaab96b$export$982f47444a2f88da),
    (0, $39dd722b123bef71$export$65235875731ff287),
    (0, $39dd722b123bef71$export$b60f30d254d1e1a6),
    (0, $39dd722b123bef71$export$25eb377ce1feaada)
];
$parcel$exportWildcard($ee8be56859ec8fb7$exports, $9a2bbe5526331901$exports);
$parcel$exportWildcard($ee8be56859ec8fb7$exports, $4c1cbb15589101a6$exports);
$parcel$exportWildcard($ee8be56859ec8fb7$exports, $2bb487cfecefdbac$exports);
$parcel$exportWildcard($ee8be56859ec8fb7$exports, $ee1dbf60aeaab96b$exports);
$parcel$exportWildcard($ee8be56859ec8fb7$exports, $39dd722b123bef71$exports);


var $6eca4c4d96d9280c$exports = {};

$parcel$export($6eca4c4d96d9280c$exports, "Editor", function () { return $6eca4c4d96d9280c$export$7cda8d932e2f33c0; });





class $6eca4c4d96d9280c$export$7cda8d932e2f33c0 extends (0, $4Pi8L$react.PureComponent) {
    static getDerivedStateFromProps({ tools: tools = (0, $ee8be56859ec8fb7$export$aa718db1466ce312) , defaultValue: defaultValue  }, { toolList: toolList , data: data  }) {
        return {
            toolList: toolList[0] ? toolList : tools.map((ToolButton)=>new ToolButton()),
            data: data || defaultValue
        };
    }
    componentDidMount() {
        document.addEventListener("selectionchange", this.updateTools);
    }
    componentWillUnmount() {
        document.removeEventListener("selectionchange", this.updateTools);
    }
    async uploadFile(Type, data) {
        if (typeof data === "string" && !/^(data|blob):/.test(data)) return data;
        const tool1 = this.state.toolList.find((tool)=>tool instanceof Type);
        try {
            return await (tool1 === null || tool1 === void 0 ? void 0 : tool1.save(typeof data === "string" ? await (0, $4Pi8L$webutility.blobOf)(data) : data));
        } catch (error) {
            console.error(error);
            if (typeof data === "string") return data;
        }
    }
    async clearHTML(markup) {
        const fragment = document.createDocumentFragment();
        fragment.append(...(0, $4Pi8L$webutility.parseDOM)(markup));
        for (const element of (0, $4Pi8L$webutility.walkDOM)(fragment, Node.ELEMENT_NODE)){
            const { style: style , dataset: dataset  } = element;
            if ([
                "nowrap",
                "pre"
            ].includes(style.whiteSpace)) style.whiteSpace = "normal";
            for(const key in dataset)delete dataset[key];
            if (element instanceof HTMLImageElement) element.src = await this.uploadFile((0, $ee1dbf60aeaab96b$export$982f47444a2f88da), element.src);
            else if (element instanceof HTMLAudioElement) element.src = await this.uploadFile((0, $ee1dbf60aeaab96b$export$6024938f37e646f8), element.src);
            else if (element instanceof HTMLVideoElement) element.src = await this.uploadFile((0, $ee1dbf60aeaab96b$export$932111b8028f79ed), element.src);
        }
        return fragment;
    }
    render() {
        const { toolList: toolList , data: data  } = this.state, { onChange: onChange  } = this.props;
        return /*#__PURE__*/ (0, $4Pi8L$reactjsxruntime.jsxs)((0, $4Pi8L$reactjsxruntime.Fragment), {
            children: [
                /*#__PURE__*/ (0, $4Pi8L$reactjsxruntime.jsx)("header", {
                    children: toolList.map((tool)=>tool.render(this.box))
                }),
                /*#__PURE__*/ (0, $4Pi8L$reactjsxruntime.jsx)("div", {
                    ref: this.box,
                    className: "form-control h-auto",
                    contentEditable: true,
                    dangerouslySetInnerHTML: {
                        __html: data
                    },
                    onPaste: this.handlePasteDrop,
                    onDrop: this.handlePasteDrop,
                    onInput: ({ currentTarget: { innerHTML: innerHTML  }  })=>{
                        return onChange === null || onChange === void 0 ? void 0 : onChange(innerHTML);
                    }
                })
            ]
        });
    }
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "box", /*#__PURE__*/ (0, $4Pi8L$react.createRef)());
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "state", {
            toolList: [],
            data: ""
        });
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "updateTools", ()=>{
            if (this.box.current === document.activeElement) this.setState({
                toolList: [
                    ...this.state.toolList
                ]
            });
        });
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "handlePasteDrop", async (event)=>{
            var _props, ref;
            event.preventDefault();
            const { currentTarget: currentTarget  } = event;
            const list = event.type === "paste" ? [
                ...event.clipboardData.items
            ] : [
                ...event.dataTransfer.items
            ];
            for (const item of list)if (item.type === "text/html") {
                const raw = await new Promise((resolve)=>item.getAsString(resolve));
                (0, $4Pi8L$webutility.insertToCursor)(await this.clearHTML(raw));
            } else if (item.type.startsWith("image/")) {
                const URI = await this.uploadFile((0, $ee1dbf60aeaab96b$export$982f47444a2f88da), item.getAsFile());
                if (URI) (0, $4Pi8L$webutility.insertToCursor)(...(0, $4Pi8L$webutility.parseDOM)(`<img src="${URI}" />`));
            } else if (item.type.startsWith("audio/")) {
                const URI = await this.uploadFile((0, $ee1dbf60aeaab96b$export$6024938f37e646f8), item.getAsFile());
                if (URI) (0, $4Pi8L$webutility.insertToCursor)(...(0, $4Pi8L$webutility.parseDOM)(`<audio src="${URI}" />`));
            } else if (item.type.startsWith("video/")) {
                const URI = await this.uploadFile((0, $ee1dbf60aeaab96b$export$932111b8028f79ed), item.getAsFile());
                if (URI) (0, $4Pi8L$webutility.insertToCursor)(...(0, $4Pi8L$webutility.parseDOM)(`<video src="${URI}" />`));
            }
            (ref = (_props = this.props).onChange) === null || ref === void 0 ? void 0 : ref.call(_props, currentTarget.innerHTML);
        });
    }
}
(0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))($6eca4c4d96d9280c$export$7cda8d932e2f33c0, "displayName", "Editor");


$parcel$exportWildcard(module.exports, $783b6726ddd1ea56$exports);
$parcel$exportWildcard(module.exports, $ee8be56859ec8fb7$exports);
$parcel$exportWildcard(module.exports, $6eca4c4d96d9280c$exports);
