// edkit/dist/index.js
var $4Pi8L$browserfsaccess = require("browser-fs-access");
var $4Pi8L$swchelperslib_define_propertyjs = require("@swc/helpers/lib/_define_property.js");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
var $5ddc77f84d52e9cd$exports = {};

$parcel$export($5ddc77f84d52e9cd$exports, "getAnchorElement", function () { return $5ddc77f84d52e9cd$export$ac81f1ba58a5f363; });
$parcel$export($5ddc77f84d52e9cd$exports, "Tool", function () { return $5ddc77f84d52e9cd$export$724f6b867059ef5f; });
$parcel$export($5ddc77f84d52e9cd$exports, "FileTool", function () { return $5ddc77f84d52e9cd$export$de3122b2484864a7; });

function $5ddc77f84d52e9cd$export$ac81f1ba58a5f363() {
    const { anchorNode: anchorNode  } = self.getSelection() || {};
    if (anchorNode) return anchorNode.parentElement;
}
class $5ddc77f84d52e9cd$export$724f6b867059ef5f {
    get usable() {
        const { command: command  } = this;
        return !command || document.queryCommandSupported(command) || document.queryCommandEnabled(command);
    }
    getActive() {
        const { command: command , tags: tags  } = this;
        if (tags) {
            const box = $5ddc77f84d52e9cd$export$ac81f1ba58a5f363();
            return !!box && box.matches(tags.map((tag)=>`${tag}, ${tag} *`).join(", "));
        }
        if (command) return document.queryCommandState(command);
        return false;
    }
    get active() {
        return this.getActive();
    }
    get title() {
        const { name: name , keys: keys , usable: usable  } = this;
        return `${name}${usable ? keys ? `\n(${keys.join(" + ")})` : "" : "\n(not supported)"}`;
    }
    edit(editor, ...data) {
        editor.focus();
        document.execCommand(this.command, null, ...data);
    }
    execute(editor, ...data) {
        var { inputs: inputs  } = this, values = [];
        if (inputs) {
            const data = inputs.map((input)=>self.prompt(input));
            if (!data.filter(Boolean)[0]) return;
            values = data;
        }
        this.edit(editor, ...values);
    }
}
class $5ddc77f84d52e9cd$export$de3122b2484864a7 extends $5ddc77f84d52e9cd$export$724f6b867059ef5f {
    async save(data) {
        return URL.createObjectURL(data);
    }
    codeOf(path) {
        return path;
    }
    async execute(editor) {
        try {
            var path = await this.save(await (0, $4Pi8L$browserfsaccess.fileOpen)());
        } catch  {
            var path = self.prompt("No file uploaded, please input an URL");
        }
        if (path) this.edit(editor, this.codeOf(path));
    }
}


var $9a2bbe5526331901$exports = {};

$parcel$export($9a2bbe5526331901$exports, "BoldTool", function () { return $9a2bbe5526331901$export$10056a49e3b218eb; });
$parcel$export($9a2bbe5526331901$exports, "ItalicTool", function () { return $9a2bbe5526331901$export$edf99d0f51bacb1e; });
$parcel$export($9a2bbe5526331901$exports, "UnderlineTool", function () { return $9a2bbe5526331901$export$4b6d9c17188960d2; });
$parcel$export($9a2bbe5526331901$exports, "StrikeThroughTool", function () { return $9a2bbe5526331901$export$a35bffaeae8fdda0; });
$parcel$export($9a2bbe5526331901$exports, "H1Tool", function () { return $9a2bbe5526331901$export$30e8c6b428f8749f; });
$parcel$export($9a2bbe5526331901$exports, "H2Tool", function () { return $9a2bbe5526331901$export$58443375efa91029; });
$parcel$export($9a2bbe5526331901$exports, "H3Tool", function () { return $9a2bbe5526331901$export$75d24d6b8b747307; });
$parcel$export($9a2bbe5526331901$exports, "FontSizeDownTool", function () { return $9a2bbe5526331901$export$a20806c6ef6938f8; });
$parcel$export($9a2bbe5526331901$exports, "FontSizeUpTool", function () { return $9a2bbe5526331901$export$744da7e24e4a7382; });
$parcel$export($9a2bbe5526331901$exports, "SubscriptTool", function () { return $9a2bbe5526331901$export$6137518b4e998008; });
$parcel$export($9a2bbe5526331901$exports, "SuperscriptTool", function () { return $9a2bbe5526331901$export$7217f06c61ebd4fc; });
$parcel$export($9a2bbe5526331901$exports, "LinkTool", function () { return $9a2bbe5526331901$export$f6c54364b5332e66; });


class $9a2bbe5526331901$export$10056a49e3b218eb extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Bold");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "b",
            "strong"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "keys", [
            "Ctrl",
            "B"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "bold");
    }
}
class $9a2bbe5526331901$export$edf99d0f51bacb1e extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Italic");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "i",
            "em"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "keys", [
            "Ctrl",
            "I"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "italic");
    }
}
class $9a2bbe5526331901$export$4b6d9c17188960d2 extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Underline");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "u"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "keys", [
            "Ctrl",
            "U"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "underline");
    }
}
class $9a2bbe5526331901$export$a35bffaeae8fdda0 extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Strike through");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "s",
            "del",
            "strike"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "strikeThrough");
    }
}
const $9a2bbe5526331901$var$HeadingCommand = document.queryCommandSupported("heading") ? "heading" : "formatBlock";
class $9a2bbe5526331901$export$30e8c6b428f8749f extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    execute(editor) {
        this.edit(editor, "H1");
    }
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "H1");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "h1"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", $9a2bbe5526331901$var$HeadingCommand);
    }
}
class $9a2bbe5526331901$export$58443375efa91029 extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    execute(editor) {
        this.edit(editor, "H2");
    }
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "H2");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "h2"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", $9a2bbe5526331901$var$HeadingCommand);
    }
}
class $9a2bbe5526331901$export$75d24d6b8b747307 extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    execute(editor) {
        this.edit(editor, "H3");
    }
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "H3");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "h3"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", $9a2bbe5526331901$var$HeadingCommand);
    }
}
class $9a2bbe5526331901$export$a20806c6ef6938f8 extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Font Size down");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "small"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "decreaseFontSize");
    }
}
class $9a2bbe5526331901$export$744da7e24e4a7382 extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Font Size up");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "big"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "increaseFontSize");
    }
}
class $9a2bbe5526331901$export$6137518b4e998008 extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Subscript");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "sub"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "subscript");
    }
}
class $9a2bbe5526331901$export$7217f06c61ebd4fc extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Superscript");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "sup"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "superscript");
    }
}
class $9a2bbe5526331901$export$f6c54364b5332e66 extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Link");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "a"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "inputs", [
            "Path"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "createLink");
    }
}


var $ceabaccd1f504406$exports = {};

$parcel$export($ceabaccd1f504406$exports, "ColorTool", function () { return $ceabaccd1f504406$export$ba9bbe5370f1bf2f; });
$parcel$export($ceabaccd1f504406$exports, "ForeColorTool", function () { return $ceabaccd1f504406$export$cab0e15cfb8489d7; });
$parcel$export($ceabaccd1f504406$exports, "BackColorTool", function () { return $ceabaccd1f504406$export$92bb6183e7459be2; });


class $ceabaccd1f504406$export$ba9bbe5370f1bf2f extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    getColor() {
        const box = (0, $5ddc77f84d52e9cd$export$ac81f1ba58a5f363)(), { colorName: colorName  } = this;
        if (!box) return colorName === "color" ? "#000000" : "#ffffff";
        const { [colorName]: color  } = getComputedStyle(box);
        const [red, green, blue] = color.match(/\d+/g);
        return "#" + (+red).toString(16).padStart(2, "0") + (+green).toString(16).padStart(2, "0") + (+blue).toString(16).padStart(2, "0");
    }
    get active() {
        return !!this.getColor();
    }
    execute(editor, color) {
        this.edit(editor, color);
    }
}
class $ceabaccd1f504406$export$cab0e15cfb8489d7 extends $ceabaccd1f504406$export$ba9bbe5370f1bf2f {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Fore Color");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "foreColor");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "colorName", "color");
    }
}
class $ceabaccd1f504406$export$92bb6183e7459be2 extends $ceabaccd1f504406$export$ba9bbe5370f1bf2f {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Back Color");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "backColor");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "colorName", "backgroundColor");
    }
}


var $2bb487cfecefdbac$exports = {};

$parcel$export($2bb487cfecefdbac$exports, "AlignTool", function () { return $2bb487cfecefdbac$export$6c1211091eb3b68f; });
$parcel$export($2bb487cfecefdbac$exports, "AlignLeftTool", function () { return $2bb487cfecefdbac$export$42a444ebfdcfe7b9; });
$parcel$export($2bb487cfecefdbac$exports, "AlignCenterTool", function () { return $2bb487cfecefdbac$export$9a2e6e0294a984eb; });
$parcel$export($2bb487cfecefdbac$exports, "AlignRightTool", function () { return $2bb487cfecefdbac$export$22363f91860fc258; });
$parcel$export($2bb487cfecefdbac$exports, "AlignFullTool", function () { return $2bb487cfecefdbac$export$42827b1362fed748; });
$parcel$export($2bb487cfecefdbac$exports, "OrderedListTool", function () { return $2bb487cfecefdbac$export$faf4e70f09f43a15; });
$parcel$export($2bb487cfecefdbac$exports, "UnorderedListTool", function () { return $2bb487cfecefdbac$export$cf4030b57a3fbe37; });
$parcel$export($2bb487cfecefdbac$exports, "HorizontalRuleTool", function () { return $2bb487cfecefdbac$export$dce8a7d6279fc201; });


class $2bb487cfecefdbac$export$6c1211091eb3b68f extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    get active() {
        if (super.getActive()) return true;
        const box = (0, $5ddc77f84d52e9cd$export$ac81f1ba58a5f363)();
        return !!box && getComputedStyle(box).textAlign === this.align;
    }
}
class $2bb487cfecefdbac$export$42a444ebfdcfe7b9 extends $2bb487cfecefdbac$export$6c1211091eb3b68f {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Align Left");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "justifyLeft");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "align", "left");
    }
}
class $2bb487cfecefdbac$export$9a2e6e0294a984eb extends $2bb487cfecefdbac$export$6c1211091eb3b68f {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Align Center");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "justifyCenter");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "align", "center");
    }
}
class $2bb487cfecefdbac$export$22363f91860fc258 extends $2bb487cfecefdbac$export$6c1211091eb3b68f {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Align Right");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "justifyRight");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "align", "right");
    }
}
class $2bb487cfecefdbac$export$42827b1362fed748 extends $2bb487cfecefdbac$export$6c1211091eb3b68f {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Align Full");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "justifyFull");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "align", "justify");
    }
}
class $2bb487cfecefdbac$export$faf4e70f09f43a15 extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Ordered list");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "ol"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "insertOrderedList");
    }
}
class $2bb487cfecefdbac$export$cf4030b57a3fbe37 extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Unordered list");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "ul"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "insertUnorderedList");
    }
}
class $2bb487cfecefdbac$export$dce8a7d6279fc201 extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Horizontal rule");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "hr"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "insertHorizontalRule");
    }
}


var $ee1dbf60aeaab96b$exports = {};

$parcel$export($ee1dbf60aeaab96b$exports, "IFrameTool", function () { return $ee1dbf60aeaab96b$export$3e557574863db086; });
$parcel$export($ee1dbf60aeaab96b$exports, "ImageTool", function () { return $ee1dbf60aeaab96b$export$982f47444a2f88da; });
$parcel$export($ee1dbf60aeaab96b$exports, "AudioTool", function () { return $ee1dbf60aeaab96b$export$6024938f37e646f8; });
$parcel$export($ee1dbf60aeaab96b$exports, "VideoTool", function () { return $ee1dbf60aeaab96b$export$932111b8028f79ed; });


class $ee1dbf60aeaab96b$export$3e557574863db086 extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    execute(editor) {
        const path = self.prompt("Path");
        if (path) this.edit(editor, `<iframe style="width: 100%; height: 50vh; border: none" src="${path}"></iframe>`);
    }
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Embed Web page");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "iframe"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "insertHTML");
    }
}
class $ee1dbf60aeaab96b$export$982f47444a2f88da extends (0, $5ddc77f84d52e9cd$export$de3122b2484864a7) {
    codeOf(path) {
        return `<img style="max-width: 100%; height: auto" src="${path}"/>`;
    }
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Image");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "img"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "insertHTML");
    }
}
class $ee1dbf60aeaab96b$export$6024938f37e646f8 extends (0, $5ddc77f84d52e9cd$export$de3122b2484864a7) {
    codeOf(path) {
        return `<audio controls src="${path}"></audio>`;
    }
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Audio");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "audio"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "insertHTML");
    }
}
class $ee1dbf60aeaab96b$export$932111b8028f79ed extends (0, $5ddc77f84d52e9cd$export$de3122b2484864a7) {
    codeOf(path) {
        return `<video controls src="${path}"></video>`;
    }
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Video");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "tags", [
            "video"
        ]);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "insertHTML");
    }
}


var $39dd722b123bef71$exports = {};

$parcel$export($39dd722b123bef71$exports, "UndoTool", function () { return $39dd722b123bef71$export$65235875731ff287; });
$parcel$export($39dd722b123bef71$exports, "RedoTool", function () { return $39dd722b123bef71$export$b60f30d254d1e1a6; });
$parcel$export($39dd722b123bef71$exports, "ResetTool", function () { return $39dd722b123bef71$export$6f46dc3d81fbb4bd; });
$parcel$export($39dd722b123bef71$exports, "ClearTool", function () { return $39dd722b123bef71$export$25eb377ce1feaada; });


class $39dd722b123bef71$export$65235875731ff287 extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Undo");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "undo");
    }
}
class $39dd722b123bef71$export$b60f30d254d1e1a6 extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Redo");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "redo");
    }
}
class $39dd722b123bef71$export$6f46dc3d81fbb4bd extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Reset");
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "command", "removeFormat");
    }
}
class $39dd722b123bef71$export$25eb377ce1feaada extends (0, $5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    execute(editor) {
        editor.focus();
        document.execCommand("selectAll");
        document.execCommand("delete");
    }
    constructor(...args){
        super(...args);
        (0, ($parcel$interopDefault($4Pi8L$swchelperslib_define_propertyjs)))(this, "name", "Clear");
    }
}


$parcel$exportWildcard(module.exports, $5ddc77f84d52e9cd$exports);
$parcel$exportWildcard(module.exports, $9a2bbe5526331901$exports);
$parcel$exportWildcard(module.exports, $ceabaccd1f504406$exports);
$parcel$exportWildcard(module.exports, $2bb487cfecefdbac$exports);
$parcel$exportWildcard(module.exports, $ee1dbf60aeaab96b$exports);
$parcel$exportWildcard(module.exports, $39dd722b123bef71$exports);
